api/net/Middleware/ErrorHandlingMiddleware.cs

```diff
@@ -99,7 +99,10 @@ private async Task HandleExceptionAsync(HttpContext context, Exception ex)
             else if (ex is DbUpdateException)
             {
                 code = HttpStatusCode.BadRequest;
-                message = "A database error occurred while updating.";
+                if (ex.InnerException?.Message.Contains("23505: duplicate key value violates unique constraint") == true)
```

Comment:
I don't love manually determining a relevant error message, but I can't think of anything more secure for production.  We don't want raw errors getting out to users.